pkg/preparer/orchestrate.go

```diff
@@ -418,7 +420,9 @@ func (p *Preparer) installAndLaunchPod(pair ManifestPair, pod Pod, logger loggin
 			}
 		} else {
 			// TODO: do this in a transaction
-			err = p.podStore.WriteRealityIndex(pair.PodUniqueKey, p.node)
+			// TODO dai - what do I do with the errors?
+			ctx, cancelFunc := transaction.New(context.Background())
```

Comment:
you should add `defer cancelFunc()` here. sorry it's kind of a janky calling pattern that maybe we should rethink.

The reasoning is that if you return (e.g. on line 440) the transaction will never be closed, and that can cause a memory leak because contexts keep some state around that isn't released until the transaction is canceled. See https://golang.org/pkg/context/#WithCancel